### PR TITLE
fix: add SearchOnSealed unit test and fix a bug

### PR DESCRIPTION
**Pull request by sunby**
issue: https://github.com/milvus-io/milvus/issues/37244

### CODE REVIEW

File: internal/core/src/query/SearchOnSealed.cpp [hunk 1]
```diff
@@ -130,30 +130,35 @@ SearchOnSealed(const Schema& schema,
                 chunk_size);
             bitset_ptr = reinterpret_cast<const uint8_t*>(bitset_data);
         }
-        offset += chunk_size;
         BitsetView bitset_view(bitset_ptr, chunk_size);
 
         if (search_info.group_by_field_id_.has_value()) {
             auto sub_qr = BruteForceSearchIterators(dataset,
```
[the remark this answers is not in the document]
Comment (sunby) [reply internal/core/src/query/SearchOnSealed.cpp:136]: > Does iterators handle different chunk_size and segment offsets?

will add test in next pr.

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**